src/sinks/aws_sqs.rs

```diff
@@ -301,13 +333,33 @@ mod tests {
         let message = "hello world".to_string();
         let mut event = Event::from(message.clone());
         event.as_mut_log().insert("key", "value");
-        let event = encode_event(event, &Encoding::Json.into(), None).unwrap();
+        let event = encode_event(event, &Encoding::Json.into(), None, None).unwrap();
 
         let map: BTreeMap<String, String> = serde_json::from_str(&event.item.message_body).unwrap();
 
         assert_eq!(map[&log_schema().message_key().to_string()], message);
         assert_eq!(map["key"], "value".to_string());
     }
+
+    #[test]
+    fn sqs_encode_event_deduplication_id() {
+        let message = "hello world".to_string();
+        let message_deduplication_id = Template::try_from("{{ transaction_id }}").unwrap();
+        let mut event = Event::from(message.clone());
+        event.as_mut_log().insert("transaction_id", "some id");
+        let event = encode_event(
+            event,
+            &Encoding::Json.into(),
+            None,
+            Some(&message_deduplication_id),
+        )
```

Comment:
I suggest you convert this into an `Event` as the last step:
```suggestion
        let mut event = LogEvent::from(message);
        event.insert(transaction_id", "some id");
        let event = encode_event(
            event.into(),
            &Encoding::Json.into(),
            None,
            Some(&message_deduplication_id),
        )
```